torchserve/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,123 @@
+name: TorchServe
+files:
+- name: torchserve.yaml
+  options:
+  - template: init_config
+    options:
+    - template: init_config/openmetrics
+    - template: init_config/http
+      overrides:
+        proxy.hidden: true
+        skip_proxy.hidden: true
+        timeout.hidden: true
+    - template: init_config/default
+      overrides:
+        service.hidden: true
+  - template: instances
+    multiple_instances_defined: true
+    options:
+      - name: OpenMetrics
+        description: | 
+          TorchServe OpenMetrics endpoint configuration
+        options:
+          - template: instances/openmetrics
+            overrides:
+              openmetrics_endpoint.required: false
+              openmetrics_endpoint.value.example: http://localhost:8082/metrics
+              exclude_labels.value.example:
+                - hostname
+                - Hostname
+              rename_labels.value.example:
+                Level: level
+                ModelName: model_name
+                WorkerName: worker_name
+              exclude_metrics_by_labels.description: |
+                A mapping of labels to exclude metrics with matching label name and their corresponding metric values. 
+                To match all values of a label, set it to `true`.
+    
+                Note: Label filtering happens before `rename_labels`.
+    
+                For example, the following configuration instructs the check to exclude all metrics with
+                a label `worker` or a label `pid` with the value of either `23` or `42`.
+    
+                  exclude_metrics_by_labels:
+                    worker: true
+                    pid:
+                    - '23'
+                    - '42'
+              cache_shared_labels.description: |
+                When `share_labels` is set, it instructs the check to cache labels collected 
+                from the first payload for improved performance.
+            
+                Set this to `false` to compute label sharing for every payload at the risk 
+                of potentially increased memory usage.
+      - name: Inference API
+        description: TorchServe Inference API endpoint configuration
+        options:
+          - name: inference_api_url
+            required: false
+            description: The TorchServe Inference API url to connect to.
+            value:
+              example: "http://localhost:8080"
+              type: string
+          - template: instances/http
```

Comment:
> Do we need multiple template: instances/http and template: instances/default for the spec.yaml? It looks like this causes the conf.yaml.example to be very long and contain multiple duplicates (i.e. ## @param auth_type) even though users cannot configure multiple auth_type for each instance anyways. The main concern I have is that some of the useful config options (inference_api_url and management_api_url) end up deep in the middle of the file, which makes it difficult to find.

Yes because we actually have 3 different sections that are different, one for OM, one for inference and one for management, even if they have some options in common, I'm not sure we can inherit from another section. 

I think this also answers the other question